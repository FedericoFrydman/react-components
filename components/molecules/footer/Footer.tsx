import { Ref, forwardRef } from 'react';
import cn from 'classnames';

import './footer.scss';

export type FooterAlignment = 'left' | 'right' | 'center';

export interface FooterProps extends Omit<React.HTMLProps<HTMLDivElement>, 'css'> {
  /**
   * Name of the "brand" on the copyright
   */
  brandName?: string;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLDivElement>;
  /**
   * HTML content.
   */
  htmlContent?: string;
  /**
   * Alignment for the footer.
   */
  align?: FooterAlignment;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
}

const FooterComponent = (
  { brandName = null, htmlContent = null, classNames = [], align = 'right', ...props }: FooterProps,
  ref: FooterProps['ref'],
) => {
  const classFooter = 'lsux-footer';
  const cnFooter = cn(
    {
      [`${classFooter}`]: true,
      [`${classFooter}-${align}`]: align,
    },
    classNames,
  );

  return (
    <div className={cnFooter} ref={ref} {...props}>
      {htmlContent && <span dangerouslySetInnerHTML={{ __html: htmlContent }} />}
      {brandName && (
        <>
          &copy; {brandName} {new Date().getFullYear()}
        </>
      )}
    </div>
  );
};

export const Footer = forwardRef<HTMLDivElement, FooterProps>(FooterComponent);
