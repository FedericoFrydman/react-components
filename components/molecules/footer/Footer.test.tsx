import { createRef } from 'react';
import { render, screen } from '@testing-library/react';
import { shallow, ShallowWrapper } from 'enzyme';
import cn from 'classnames';

import { Footer, FooterProps } from './Footer';

describe('Footer Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer />);
    testId = 'test-footer';
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the image the provided classNames', () => {
          const image = renderImage({ classNames });

          expect(image).toHaveClass(cn(classNames), { exact: false });
        });
      });
    });
  });

  describe('ref', () => {
    const ref = createRef<HTMLDivElement>();

    it('provides a ref to the image', () => {
      expect(renderImage({ ref })).toEqual(ref.current);
    });
  });

  const renderImage = (props: FooterProps = {}): HTMLElement => {
    render(<Footer {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});
