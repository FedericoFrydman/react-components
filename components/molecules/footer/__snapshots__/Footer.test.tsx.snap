// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render the component 1`] = `
<div
  className="lsux-footer lsux-footer-right"
/>
`;
